### PR TITLE
[release/1.0.x] Onboard to CRT

### PR DESCRIPTION
**Pull request by claire-labry**
Onboard Nomad OSS [release/1.0.x] to Common Release Tooling

There are a few important changes to call out in this PR:

GitHub Actions is being used as the CI provider to build the go binary artifacts and linux packages. This logic lives in .github/workflows/build.yml and workflow runs are available under https://github.com/hashicorp/nomad/actions.
Artifact builds will continue to run on PR's and merges to main. This behavior can be changed by updating the on: logic in the .github/workflows/build.yml file.

The .release/ci.hcl file contains some metadata for our release service. The only part you'll likely ever need to change is the release_branches array -- when you want to release off of a new release branch, update this value to the release branch name and be sure to update the branches array in .github/workflows/build.yml to the same value. This should be set to either main, the release branch that we'll be releasing off of (once release branches are adopted), or a test branch that you are working off of if you need to validate every part of the pipeline before merging.

- [x]  Need to figure out why `security-scan-binaries` is failing - might require an update to `bob`
- [x] Clean up this branch before merging [slack ID and branch naming from onboarding tests] 

This PR includes the Generate Static Assets workflow-- this workflow generates the static files that are needed for release day. It commits any generated static files that were made into the release branch. These files will either be committed (if files are generated) or no files will be pushed based on the changes that are pushed up. 

- [generate-static-assets workflow run passed here:](https://github.com/hashicorp/nomad/actions/runs/2098144903)

- This workflow then kicks off the [build workflow which passed here:](https://github.com/hashicorp/nomad/actions/runs/2098167494)

- You can see [here](https://github.com/hashicorp/nomad/runs/5839297365?check_suite_focus=true#step:11:22) that there are files that were updated (committed).

image

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.16.5
```
Comment (tgross) [inline .go-version:1]: Is this version used for building the binary itself? Nomad has to be built with go 1.17+.

Comment (mdeggies) [reply .go-version]: Yes, `.go-version` is where the go-version is set and used for all CI build jobs. I just updated this to 1.17.5. Thanks for the note!

Comment (mdeggies) [reply .go-version]: Just wanted to confirm this is needed for this older release branch. go mod shows Go 1.16, and the builds succeed with Go 1.16.5 but not 1.17.5. I can dig into this more next week, but just want to clarify that this branch was previously building with 1.17.5 before spending the time on it.

Comment (tgross) [reply .go-version]: Oh shoot, I'm sorry. I wasn't paying close enough attention and didn't realize this was a separate PR for 1.0.x. You're right, that's on 1.16 still. 🤦

Comment (mdeggies) [reply .go-version:1]: No worries! I'll revert back to 1.16 for now.

**Review by kpenfound — APPROVED** [review]
🚢

**Review by sarahethompson — APPROVED** [review]

**Review by lgfa29 — APPROVED** [review]

**Pull request merged**